src/backend/distributed/commands/table.c

```diff
@@ -182,6 +200,12 @@ PostprocessCreateTableStmt(CreateStmt *createStatement, const char *queryString)
 	{
 		ErrorOutForFKeyBetweenPostgresAndCitusLocalTable(relationId);
 	}
+
+	/* invalidate foreign key cache if the table involved in any foreign key */
+	if ((TableReferenced(relationId) || TableReferencing(relationId)))
```

Comment:
In general, my only remaining concern on this PR is that we call `InvalidateForeignKeyGraph ()` for PG tables only. So, for example if an unrelated job on the coordinator consistently creates/drops foreign keys on unrelated local tables, we'll be unnecessarily re-creating the graph on all backends.

Is there a way to improve that? For example, can we have a variant of `TableReferenced / TableReferencing `, which returns false if the tables has no interaction with distributed tables?